app/src/main/java/com/cmput301f20t21/bookfriends/ui/library/OwnedListFragment.java

```diff
@@ -103,10 +105,10 @@ public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceStat
         recyclerView.setLayoutManager(layoutManager);
 
         // set a temporary adapter
-        recyclerView.setAdapter(new OwnedListAdapter(new ArrayList<>(), this::onItemClick, this::onDeleteBook));
+        recyclerView.setAdapter(new OwnedListAdapter(new ArrayList<>(), this::onItemClick, this::onDeleteBook, this::onViewRequests));
 
         vm.getBooks().observe(getViewLifecycleOwner(), (List<Book> books) -> {
-            mAdapter = new OwnedListAdapter(books, this::onItemClick, this::onDeleteBook);
+            mAdapter = new OwnedListAdapter(books, this::onItemClick, this::onDeleteBook, this::onViewRequests);
```

Comment:
@c25vdw for adapters, should we still pass in callbacks like this?

Reply:
we are passing item listeners from fragment to adapter and adapter to items....It's kinda messy but ok if we only have these listeners and no more, but let me see if there's a better way of doing this